src/Text/KirbyTag.php

```diff
@@ -116,35 +116,29 @@ public static function factory(...$arguments): string
 	 */
 	public function file(string $path): File|null
 	{
-		$parent = $this->parent();
-
-		// check first for UUID
-		if (Uuid::is($path, 'file') === true) {
-			if (
-				is_object($parent) === true &&
-				method_exists($parent, 'files') === true
-			) {
+		if ($parent = $this->parent()) {
+			// check first for UUID
+			if (method_exists($parent, 'files') === true) {
 				$context = $parent->files();
 			}
 
-			return Uuid::for($path, $context ?? null)->model();
-		}
+			if ($uuid = Uuid::from($path, 'file', $context ?? null)) {
+				return $uuid->model();
+			}
 
-		if (
-			is_object($parent) === true &&
-			method_exists($parent, 'file') === true &&
-			$file = $parent->file($path)
-		) {
-			return $file;
-		}
+			if (method_exists($parent, 'file') === true) {
+				return $parent->file($path);
```

Comment:
`Issue`: If called on `$parent = Page`, this will return `null` if that page doesn't have the file `$path`. The old code continued to the very last line of the method and checked for site files of that name.